recipes/omniscidb-cuda/recipe/build.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -ex
+
+# Make sure -fPIC is not in CXXFLAGS (that some conda packages may
+# add), otherwise omniscidb server will crash when executing generated
+# machine code:
+export CXXFLAGS="`echo $CXXFLAGS | sed 's/-fPIC//'`"
+
+# Fixes https://github.com/Quansight/pearu-sandbox/issues/7
+#       https://github.com/omnisci/omniscidb/issues/374
+export CXXFLAGS="$CXXFLAGS -Dsecure_getenv=getenv"
+
+# Fixes `undefined reference to
+# `boost::system::detail::system_category_instance'`:
+export CXXFLAGS="$CXXFLAGS -DBOOST_ERROR_CODE_HEADER_ONLY"
+
+# Remove --as-needed to resolve undefined reference to `__vdso_clock_gettime@GLIBC_PRIVATE'
+export LDFLAGS="`echo $LDFLAGS | sed 's/-Wl,--as-needed//'`"
+export LDFLAGS="$LDFLAGS -lresolv -pthread -lrt"
+
+export CC=${PREFIX}/bin/clang
+export CXX=${PREFIX}/bin/clang++
+export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DCMAKE_C_COMPILER=${CC} -DCMAKE_CXX_COMPILER=${CXX}"
+export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DCUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME"
+
+# Required for building on a CUDA enabled system without libcuda.so.1
+export EXTRA_CMAKE_OPTIONS="$EXTRA_CMAKE_OPTIONS -DCMAKE_LIBRARY_PATH=$CUDA_HOME/lib64/stubs"
+
+mkdir -p build
+cd build
+
+# TODO: when building on a system with no GPUs, using
+# -DENABLE_TESTS=off will save build time
+
+cmake -Wno-dev \
+    -DCMAKE_PREFIX_PATH=$PREFIX \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DCMAKE_BUILD_TYPE=release \
+    -DMAPD_DOCS_DOWNLOAD=off \
+    -DENABLE_AWS_S3=off \
+    -DENABLE_CUDA=on \
+    -DENABLE_FOLLY=off \
+    -DENABLE_JAVA_REMOTE_DEBUG=off \
+    -DENABLE_PROFILER=off \
+    -DENABLE_TESTS=on \
+    -DPREFER_STATIC_LIBS=off \
+    $EXTRA_CMAKE_OPTIONS \
+    ..
+
+make -j $CPU_COUNT
+
+make install
+
+# skip tests when libcuda.so is not available
+if [ "`ldd bin/initdb | grep "not found" | tr -d '[:space:]'`" == "libcuda.so.1=>notfound" ]; then
+    echo "SKIP RUNNING SANITY TESTS: libcuda.so.1 not found"
+else
+    # Running sanity tests requires CUDA enabled system with GPU(s).
+
+    # Add g++ include paths for astparser. This is used by the
+    # loadtime UDF support, resolves not found include files errors:
+    # cstdint, bits/c++config.h, features.h
+    GCCVERSION=$(basename $(dirname $($GXX -print-libgcc-file-name)))
+    export CPLUS_INCLUDE_PATH=$PREFIX/$HOST/include/c++/$GCCVERSION:$PREFIX/lib/gcc/$HOST/$GCCVERSION/include
```

Comment:
If you are adding this, then all the include paths should be added in the order that they are normally added. Run `$CXX -v` to get the order

Reply:
I am investigating this further.
Note that this part of the script does not affect the built conda package as CI instances do not have GPUs to run the omnisci sanity tests.
Currently, we have to rely on running the sanity tests locally.

Reply:
Removed CPLUS_INCLUDE_PATH as not needed.